src/admin.py

```diff
@@ -133,6 +136,54 @@ def change_password(
     db.commit()
     print(f"Password updated for user '{username}'.")
 
+@app.command()
+def create_api_key(
+    username: str = typer.Argument(None, help="User to create API key for."),
+    key_name: str = typer.Option(
+        None, "--key-name", "-n", help="Name for the API key."),
+    key_description: Optional[str] = typer.Option(
+        None, "--description", "-d", help="Description for the API key (optional)."
+    ),
+):
+    """Create an API key for a user."""
+    db = database.SessionLocal()
+
+    # Check for existing user *before* potentially prompting
+    if not username and get_user_by_username_from_db(db, username):
+        print("[bold red]Error: User already exists.[/bold red]")
+        raise typer.Exit(code=1)
+
+    if not username:
```

Comment:
How can username be None? Doesn't typer enforce that username argument is not None in the first line of this method? (line 141). afaik arguments in typer are required and options are optional by default.

Reply:
No, I set the the default if not provided to `None` so I can catch that and prompt the user for a username on the CLI.
```
username: str = typer.Argument(None, help="User to create API key for.")
```

To make the argument mandatory it would be:
```
username: str = typer.Argument(..., help="User to create API key for.")
```
Note the `...` instead of `None`.

This is typer design. A bit unintuitive IMO but it is what it is :)

Reply:
Ack. Thanks for explaining, makes sense.